Dangling indices documentation (#58751)
elastic/elasticsearch: docs/reference/modules/gateway.asciidoc

[[dangling-indices]]
==== Dangling indices

When a node joins the cluster, if it finds any shards stored in its local data
directory that do not already exist in the cluster, it will consider those
shards to be "dangling". You can automatically import dangling indices

> into the cluster using the following setting, however this is not safe, and

Reviewer: I'm a bit confused by the "this is not safe". Without further qualification, this would leave a reader puzzled as well.

> shards to be "dangling". You can automatically import dangling indices
> into the cluster using the following setting, however this is not safe, and
> you should use the <<dangling-indices-api,Dangling indices API>> instead.

Reviewer 2: ```suggestion
shards to be "dangling". Importing dangling indices
into the cluster using `gateway.auto_import_dangling_indices` is not safe. 
Instead, use the <<dangling-indices-api,Dangling indices API>>.
```

Reviewer 2: I think that this revision is more strongly worded and indicates that users should definitely avoid this setting in favor of the API.

`gateway.auto_import_dangling_indices`::

    deprecated:[7.9.0, This setting will be removed in 8.0. You should use the dedicated dangling indices API instead.]
    Whether to automatically import dangling indices into the cluster
    state, provided no indices already exist with the same name. Defaults
    to `false`.

WARNING: The auto-import functionality was intended as a best effort to help users

> who lose all master nodes. For example, if a new master node were to be

Reviewer: I'm wondering if we should add that that scenario (losing all master nodes) should be avoided at all cost, as it puts the cluster's metadata and data at risk. Also worth pointing out above (when talking about "this is not safe") that importing dangling indices does not provide any guarantees as to whether the data that's imported truly represents the latest state of the data as it was when the index was still part of the cluster.

Author: @ywelsch I've reworked the text to and address your points.

shards to be "dangling". Importing dangling indices
into the cluster using `gateway.auto_import_dangling_indices` is not safe.
Instead, use the <<dangling-indices-api,Dangling indices API>>. Neither
mechanism provides any guarantees as to whether the imported data truly
represents the latest state of the data when the index was still part of
the cluster.

`gateway.auto_import_dangling_indices`::

    deprecated:[7.9.0, This setting will be removed in 8.0. You should use the dedicated dangling indices API instead.]
    Whether to automatically import dangling indices into the cluster
    state, provided no indices already exist with the same name. Defaults
    to `false`.

WARNING: The auto-import functionality was intended as a best effort to help users
who lose all master nodes. For example, if a new master node were to be
started which was unaware of the other indices in the cluster, adding the
old nodes would cause the old indices to be imported, instead of being
deleted. However there are several issues with automatic importing, and
its use is strongly discouraged in favour of the
<<dangling-indices-api,dedicated API>.

> WARNING: Losing all master nodes is a situation that should be avoided at

Author: @lockewritesdocs Is it overkill to have two `WARNING:` sections? I felt that they were making separate points, and all the content here ought to be scary.
